File: dotnet/01-commands/doc/Banking_Kata/Account.cs

namespace Banking_Kata
{
    /// <summary>
    /// The 'Account' class represents a bank account with capabilities to track balance and transactions.
    /// </summary>
    internal class Account
    {
        public int balance { get; private set; }
        public List<BankingTransactionDTO> statement { get; private set; }

        public Account()
        {
            statement = new List<BankingTransactionDTO>();
        }

        public void Deposit(int depositAmount)
        {
            balance += depositAmount;
            statement.Add(new BankingTransactionDTO(DateTime.Today, +depositAmount, balance));
        }

        public void Withdraw(int withdrawAmount)
        {
            balance -= withdrawAmount;
            statement.Add(new BankingTransactionDTO(DateTime.Today, -withdrawAmount, balance));
        }
    }
}

File: python/01-commands/doc/Banking_Kata/account.py
from datetime import date
from banking_transaction_dto import BankingTransactionDTO

class Account:
    def __init__(self):
        self.balance = 0
        self.statement = []

    def deposit(self, deposit_amount):
        self.balance += deposit_amount
        self.statement.append(BankingTransactionDTO(date.today(), deposit_amount, self.balance))

    def withdraw(self, withdraw_amount):
        self.balance -= withdraw_amount
        self.statement.append(BankingTransactionDTO(date.today(), -withdraw_amount, self.balance))